botnim/benchmark/assistant_loop.py

```diff
@@ -64,30 +92,87 @@ def assistant_loop(client: OpenAI, assistant_id, question=None, thread=None, not
+                                text = result.content[0].text if result.content else None
+                                if text:
+                                    f.write(f"  Result:\n{text}\n")
 
-        if run.status == 'completed': 
+        if run.status == 'completed':
+            # Log assistant's response when run is completed
+            messages = client.beta.threads.messages.list(thread_id=thread.id, order='desc', limit=1)
+            for message in messages:
+                if message.role == "assistant":
+                    with open(log_file, 'a', encoding='utf-8') as f:
+                        f.write("\nAssistant Response:\n")
+                        for content in message.content:
+                            if content.type == 'text':
+                                f.write(f"{content.text.value}\n")
+                                break  # Only write the latest response
+            
+            # Log completion
+            with open(log_file, 'a', encoding='utf-8') as f:
+                f.write("\n=== Run Completed ===\n\n")
             break
 
         for tool in run.required_action.submit_tool_outputs.tool_calls:
             arguments = json.loads(tool.function.arguments)
-            if tool.function.name == 'DatasetDBQuery':
-                arguments['page_size'] = 30
-            output = get_openapi_output(openapi_spec, tool.function.name, arguments)
-            if tool.function.name == 'DatasetInfo':
+            output = None  # Initialize output variable
+            
+            # Log tool input
+            with open(log_file, 'a', encoding='utf-8') as f:
+                f.write(f"\nTool Input:\n  Tool ID: {tool.id}\n  Name: {tool.function.name}\n  Arguments:\n")
+                for key, value in arguments.items():
+                    f.write(f"    {key}: {value}\n")
+            
+            if tool.function.name.startswith('ElasticVectorSearch'):
+                tool_name = tool.function.name[len('ElasticVectorSearch_'):]
+                parts = tool_name.split('_', 1)
+                bot_name = parts[0]
+                context_name = parts[1] if len(parts) > 1 else ''
+                
+                # Log the tool call parameters
+                logger.info(f"Calling elastic_vector_search_handler with query: {arguments['query']}, num_results: {arguments.get('num_results', 7)}")
+                
+                output = elastic_vector_search_handler(
+                    environment=environment,
+                    bot_name=bot_name,
+                    context_name=context_name,
+                    query=arguments['query'],
+                    num_results=arguments.get('num_results', 7)
+                )
+                
+                # Log the output
+                logger.info(f"Tool output: {output}")
+            
+            elif tool.function.name == 'DatasetDBQuery':
+                output = get_openapi_output(openapi_spec, tool.function.name, arguments)
```

Comment:
this should not happen only for DatasetDBQuery

Reply:
fixed

Reply:
This is still not resolved - try to follow the flow in the original code and see when `get_openapi_output` is being called.

Reply:
fixed correctly this time, I hope.

Reply:
no, not yet :(
see the indentation of this line and the one above it in the original code and in yours.

Reply:
take 3

Reply:
🎉